sandman_web/sandman_web/status/status.py

```diff
@@ -0,0 +1,101 @@
+import psutil, docker, requests
+
+from flask import (
+    Blueprint, render_template
+)
+from werkzeug.exceptions import abort
+
+def process_status(process_name):
+    for process in psutil.process_iter(['pid', 'name']):
+        if process.info['name'] == process_name:
+            return True
+    return False
+
+#Check that Sandman is running
+def check_sandman_health():
+        process_name = "sandman"
+        if process_status(process_name):
+            return True
+        else:
+            return False
+
+#Check that Rhasspy is running and responding
+#Get the Rhasspy contatiner status
+def check_rhasspy_health():
+    client = docker.DockerClient(base_url='unix://var/run/docker.sock')
+    rhasspy_container = client.containers.get("rhasspy")
+    rhasspy_container_status = rhasspy_container.attrs["State"]
+
+    #Get the Rhasspy web response
+    try:
+        rhasspy_web_response = requests.get('http://localhost:12101')
+        rhasspy_web_status = rhasspy_web_response.status_code
+    except:
+        rhasspy_web_status = False
+
+    #Check that the Rhasspy container is running and the web response is OK
+    try:
+        if rhasspy_container_status["Status"] == "running" and rhasspy_web_status == 200:
+            return True
+        else:
+            return False
+    except:
+        #The container may not exist
+        return "Failed check"
+
+#Check that ha-bridge is running and responding
+#Get the ha-bridge contatiner status
+def check_ha_bridge_health():
+    client = docker.DockerClient(base_url='unix://var/run/docker.sock')
+    try:
+        ha_bridge_container = client.containers.get("ha-bridge")
+        ha_bridge_container_status = ha_bridge_container.attrs["State"]
+    except:
+        ha_bridge_container_status = False
+
+    #Get the ha-bridge web response
+    try:
+        ha_bridge_web_response = requests.get('http://localhost:80')
+        ha_bridge_web_status = ha_bridge_web_response.status_code
+    except:
+        ha_bridge_web_status = False
+
+    #Check that the ha-bridge container is running and the web response is OK
+    try:
```

Comment:
So an interesting thing about this is that I actually don't run HA bridge in a docker container. Maybe I should, but I just have never done it that way. So this would actually feel for me. I think one way to deal with this is that if you get a web response, it doesn't matter whether the container is running or not, it is running. This could technically apply to other things as well, but I don't know that there's another way to run Rhasspy beside the container.